api/source/mk_events_objects.cpp

```diff
@@ -228,6 +228,30 @@ API_EXPORT int API_CALL mk_media_source_broadcast_msg(const mk_media_source ctx,
     return src->broadcastMessage(any);
 }
 
+API_EXPORT const char* API_CALL mk_media_source_get_origin_url(const mk_media_source ctx) {
+    assert(ctx);
+    MediaSource *src = (MediaSource *)ctx;
+    return _strdup(src->getOriginUrl().c_str());
```

Comment:
`#include <string.h>`?

Reply:
Thank you, I have found the reason why the non Windows environment did not compile successfully

Reply:
Are you talking `_WIN32` inside `api/source/mk_util.cpp`? 

https://stackoverflow.com/questions/7582394/strdup-or-strdup
It seems `_strdup` is WIN standards, then it's duplicated defined here and `api/source/mk_util.cpp`, maybe extract this Macro to a utility header? (`api/include/mk_common.h`?)

Reply:
Thank you, there are indeed duplicate definitions. I have defined them in `api/include/mk_common.h`